scripts/benchmark.js

```diff
@@ -14,126 +14,181 @@ const fs     = require("fs");
+var filename = process.argv[3];
+
+if (!packagename) {
+	console.error("Error: No package specified");
+	process.exit(1);
 }
 
+console.log("package: " + packagename);
+
+const pathToScripts = "./scripts";
+
 const table = new Table({
-  head: ["", "raw", "raw win", "gzip", "gzip win", "parse time", "run"],
-  chars: {
-    top: "",
-    "top-mid": "" ,
-    "top-left": "" ,
-    "top-right": "",
-    bottom: "" ,
-    "bottom-mid": "" ,
-    "bottom-left": "" ,
-    "bottom-right": "",
-    left: "",
-    "left-mid": "",
-    mid: "",
-    "mid-mid": "",
-    right: "" ,
-    "right-mid": "",
-    middle: " ",
-  },
-  style: {
-    "padding-left": 0,
-    "padding-right": 0,
-    head: ["bold"],
-  },
+	head: ["", "raw", "raw win", "gzip", "gzip win", "parse time", "run"],
+	chars: {
+		top: "",
+		"top-mid": "" ,
+		"top-left": "" ,
+		"top-right": "",
+		bottom: "" ,
+		"bottom-mid": "" ,
+		"bottom-left": "" ,
+		"bottom-right": "",
+		left: "",
+		"left-mid": "",
+		mid: "",
+		"mid-mid": "",
+		right: "" ,
+		"right-mid": "",
+		middle: " ",
+	},
+	style: {
+		"padding-left": 0,
+		"padding-right": 0,
+		head: ["bold"],
+	},
 });
 
 let results = [];
 
-const code = fs.readFileSync(filename, "utf8");
-const gzippedCode = zlib.gzipSync(code);
-
-function test(name, callback) {
-  console.log("testing", name);
-
-  const start = Date.now();
-  const result = callback(code);
-  const end = Date.now();
-  const run = end - start;
-
-  fs.writeFileSync(".test_gen_" + name + ".js", result);
+var exec = child.exec;
+var code, gzippedCode;
+
+function installPackage() {
+	var command = "npm install --prefix " + pathToScripts + " " + packagename;
+	var installed = new Promise(function(resolve, reject) {
+		exec(command, function (error, stdout, stderr) {
+			if (error !== null) {
+				  reject('Error installing package ' + packagename + ': ' + error);
+			}
+			else {
+				// If filename has not been passed as an argument, attempt to resolve file from package.json
+				filename = filename ? pathToScripts + "/node_modules/" + filename : require.resolve(packagename);
+				if(!filename) {
+					reject("No file to test. Exiting.");
+				}
+				else {
+					console.log("file: " + filename.split(/[\\\/]/).pop());
+					resolve();
+				}
+			}
+		});
+	});
+	return installed;
+}
 
-  const gzipped = zlib.gzipSync(result);
+function uninstallPackage() {
+	exec("npm uninstall --prefix " + pathToScripts + " " + packagename, function (error, stdout, stderr) {
+		if (error !== null) {
+			console.log('Error uninstalling package ' + packagename + ': ' + error);
+		}
+	});
+}
 
-  const parseStart = Date.now();
-  // disable for now we have a failing test in dce
-  new Function(result);
-  const parseEnd = Date.now();
-  const parseNow = parseEnd - parseStart;
-
-  results.push({
-    name: name,
-    raw: result.length,
-    gzip: gzipped.length,
-    parse: parseNow,
-    run: run,
-  });
+function testFile() {
+	code = fs.readFileSync(filename, "utf8");
```

Comment:
code? Yes I guess so but it would mean passing it as an argument to other functions where it is used. The original script had it declared globally so I thought it'd be fine like that.